web/packages/teleport/src/UnifiedResources/ResourceActionButton.tsx

```diff
@@ -177,19 +206,32 @@ const AppLaunch = ({ app }: { app: App }) => {
       </ButtonBorder>
     );
   }
+  function handleSamlAppEditButtonClick() {
+    setUpdateDialogOpen(true);
+    setResourceSpec({
+      name: name,
+      event: DiscoverEventResource.SamlApplication,
+      kind: ResourceKind.SamlApplication,
+      samlMeta: { preset: samlAppPreset },
+      icon: 'Application',
+      keywords: 'saml',
+    });
```

Comment:
Hmm, why do we need two callbacks for that? Doesn't setting resource spec mean that the dialog should be open?
I'd call it `onUpdateResource: (resourceSpec: ResourceSpec) => void`.

Reply:
https://github.com/gravitational/teleport/pull/43871/commits/d7c6d0d59d77102a981c11e625dec7ac6aa0d0c5#diff-ffc6bf1267af2b53c1dad2ff62eab6a4b6ae53e7b80ce49e33b62fc5b1808e5e